src/test/java/io/cdap/plugin/dlp/UtilsTest.java

```diff
@@ -65,4 +67,42 @@ public void testTableFromStructuredRecord() throws Exception {
     Table table = Utils.getTableFromStructuredRecord(recordBuilder.build());
     Assert.assertEquals(table, tableBuilder.build());
   }
+
+  @Test(expected = ValidationException.class)
+  public void testEmptyCustomTemplate() {
+    DLPTransformPluginConfig config = new DLPTransformPluginConfig();
+    config.customTemplateEnabled = true;
+    config.templateId = null;
+    MockFailureCollector collector = new MockFailureCollector("customTemplateEnabledAndEmpty");
+    config.validate(collector, null);
+    collector.getOrThrowException();
```

Comment:
Check the validation failure details similar to what you've done in `testBothCustomTemplateEnabled()`

Reply:
PR updated accordingly.